report_dynamic/models/report_dynamic.py

```diff
@@ -0,0 +1,271 @@
+from odoo import _, api, fields, models
+from odoo.exceptions import UserError
+
+
+class ReportDynamic(models.Model):
+    _name = "report.dynamic"
+    _description = "Dynamically create reports"
+
+    name = fields.Char(required=True)
+    model_id = fields.Many2one("ir.model")
+    # Inform the user about configured model_id
+    # in template
+    model_name = fields.Char(related="model_id.name")
+    model_model = fields.Char(related="model_id.model", string="Tech name of model_id")
+    res_id = fields.Integer()
+    resource_ref = fields.Reference(
+        string="Target record",
+        selection="_selection_target_model",
+        compute="_compute_resource_ref",
+        inverse="_inverse_resource_ref",
+        store=True,
+    )
+    wrapper_report_id = fields.Many2one("ir.ui.view", domain="[('type', '=', 'qweb')]")
+    template_id = fields.Many2one(
+        "report.dynamic",
+        domain="[('is_template', '=', True)]",
+        copy=False,
+        default=lambda self: self.env.company.external_report_layout_id,
+    )
+    report_ids = fields.One2many(
+        "report.dynamic",
+        "template_id",
+        domain="[('is_template', '=', False)]",
+        copy=False,
+    )
+    documentation = fields.Text(default="Documentation placeholder", readonly=True)
+    condition_domain_global = fields.Char(
+        string="Global domain condition", default="[]"
+    )
+    active = fields.Boolean(default=True)
+    is_template = fields.Boolean(default=False)
+    lock_date = fields.Date()
+    field_ids = fields.Many2many(
+        "ir.model.fields", "contextual_field_rel", "contextual_id", "field_id", "Fields"
+    )
+    window_action_exists = fields.Boolean(compute="_compute_window_action_exists")
+    group_by_record_name = fields.Char(
+        compute="_compute_group_by_record_name",
+        store=True,
+        help="Computed field for grouping by record name in search view",
+    )
+
+    @api.model
+    def _selection_target_model(self):
+        models = self.env["ir.model"].search([])
+        return [(model.model, model.name) for model in models]
+
+    @api.depends("model_id", "res_id", "template_id")
+    def _compute_resource_ref(self):
+        for this in self:
+            model = (
+                this.model_id.model
+                if this.is_template
+                else this.template_id.model_id.model
+            )
+            if model:
```

Comment:
```suggestion
            if not model:
                rec.resource_ref = False
                continue
```
and save 1 indentation level for the whole method